### PR TITLE
Implement `EEMethodSymbol.DeclaringSyntaxReferences`

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/65165

### CODE REVIEW

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs [hunk 1]
```diff
@@ -385,7 +385,7 @@ public override ImmutableArray<Location> Locations
 
         public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
         {
-            get { throw ExceptionUtilities.Unreachable(); }
+            get { return ImmutableArray.Create(CSharpSyntaxTree.Dummy.GetRoot().GetReference()); }
```
Comment (AlekseyTs) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs:388]: >CSharpSyntaxTree.Dummy.GetRoot().GetReference() [](http://example.com/codeflow?start=47&length=47)

This implementation feels somewhat surprising. We could return an empty array, Alternatively, I think we have a helper (```GetDeclaringSyntaxReferenceHelper```?) that creates result based on locations

Comment (AlekseyTs) [reply src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs:388]: For example, see
```
        public sealed override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
        {
            get
            {
                return IsImplicitlyDeclared ?
                    ImmutableArray<SyntaxReference>.Empty :
                    GetDeclaringSyntaxReferenceHelper<ParameterSyntax>(_locations);
            }
        }
```

Comment (AlekseyTs) [reply src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs:388]: In src\Compilers\CSharp\Portable\Symbols\Source\SourceParameterSymbol.cs

**Comment by AlekseyTs** [comment]
Does VB potentially have the same issue with ```DeclaringSyntaxReferences``` implementation?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for second review. Thanks

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Symbols/EEMethodSymbol.vb [hunk 2]
```diff
@@ -359,7 +359,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
 
         Public Overrides ReadOnly Property DeclaringSyntaxReferences As ImmutableArray(Of SyntaxReference)
             Get
-                Throw ExceptionUtilities.Unreachable
+                Return GetDeclaringSyntaxReferenceHelper(Of VisualBasicSyntaxNode)(_locations)
```
Comment (RikkiGibson) [inline src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Symbols/EEMethodSymbol.vb:362]: Were we able to reproduce an analogous issue on the VB side?

**Review by RikkiGibson — APPROVED** [review]

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Symbols/EEMethodSymbol.vb (hunk 2 above)
Comment (jcouv) [reply src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/Symbols/EEMethodSymbol.vb:362]: Nope. So made the change proactively but without test

**Pull request merged**